pandas/core/groupby/generic.py

```diff
@@ -1724,13 +1731,8 @@ def _transform_item_by_item(self, obj: DataFrame, wrapper) -> DataFrame:
             output[i] = sgb.transform(wrapper)
             inds.append(i)
 
-        if not output:
-            raise TypeError("Transform function invalid for data types")
-
-        columns = obj.columns.take(inds)
```

Comment:
Looks like we can also get rid of `inds` in this method

Reply:
good catch, will update